test_launch_ros/test/test_launch_ros/test_normalize_parameters.py

```diff
@@ -212,31 +212,35 @@ def test_dictionary_with_bytes():
 
 
 def test_dictionary_with_dissimilar_array():
-    orig = [{'foo': 1, 'fiz': [True, 2.0, 3]}]
-    norm = normalize_parameters(orig)
-    expected = ({'foo': 1, 'fiz': ('True', '2.0', '3')},)
-    # assert evaluate_parameters(LaunchContext(), norm) == expected
+    with pytest.raises(TypeError) as exc:
+        orig = [{'foo': 1, 'fiz': [True, 2.0, 3]}]
```

Comment:
What happens if `orig` is `[{'foo': 1, 'fiz': ['True', '2.0', '3']}]` ? Does that get correcly interpretted as a list of strings, or does it raise `TypeError`?

Reply:
It raises `TypeError`. For avoiding it, the user should write `["'True'", "'2.0'", "'3'"]`.
I added test cases in: https://github.com/ros2/launch_ros/pull/33/commits/23421c96d9760338e7e68eee9bb37b8f4e4ea826.

I don't know if this is the best way to go or not. But I think that behavior is consistent with the other test cases I have added (in https://github.com/ros2/launch_ros/pull/33/commits/23421c96d9760338e7e68eee9bb37b8f4e4ea826).

Reply:
How so? Are plain dictionary values evaluated as YAML too? [It doesn't look like it](https://github.com/ros2/launch_ros/blob/master/launch_ros/launch_ros/utilities/evaluate_parameters.py#L88).

Reply:
Strings are converted to `TextSubstitution` in `normalize_parameters` before.
I don't have a better solution in mind, for avoiding the extra quotes.

Reply:
> How so? Are plain dictionary values evaluated as YAML too? [It doesn't look like it](https://github.com/ros2/launch_ros/blob/master/launch_ros/launch_ros/utilities/evaluate_parameters.py#L88).

Thinking about this again, I stopped converting raw strings (`str`) and `TextSubstitution` like yaml.
The second one is needed, as `str` are converted to `TextSubstitution` in `normalize_parameters`.

I'm happy with how the test looks now, but I have to recognize that both `normalize_parameters` and `evaluate_parameters` need a refactor. But I think that's ok for a follow-up.